db/cart.go

```diff
@@ -0,0 +1,82 @@
+package db
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	logger "github.com/sirupsen/logrus"
+)
+
+type Product struct {
+	Id          int     `db:"id" json:"id"`
+	Name        string  `db:"name" json:"name"`
+	Description string  `db:"description" json:"description"`
+	Price       float64 `db:"price" json:"price"`
+	Discount    float64 `db:"discount" json:"discount"`
+	Quantity    int     `db:"quantity" json:"quantity"`
+	CategoryId  int     `db:"category_id" json:"category_id"`
+}
+
+// CartProduct is rquirement of frontend as json response
+type CartProduct struct {
+	Id          int      `db:"id" json:"id"`
+	Name        string   `db:"product_title" json:"product_title"`
+	Quantity    int      `db:"quantity" json:"quantity"`
+	Category    string   `db:"category" json:"category,omitempty"`
+	Price       float64  `db:"price" json:"product_price"`
+	Description string   `db:"description" json:"description"`
+	ImageUrls   []string `db:"image_url" json:"image_url,omitempty"`
+}
+
+const (
+	getCartQuery         = `SELECT product_id FROM cart WHERE id=$1`
+	geCarttQuantityQuery = `SELECT quantity FROM cart WHERE id=$1`
+	getProductsQuery     = `SELECT * FROM products WHERE id IN (?)`
+	getCategoryQuery     = `SELECT name from category where id=$1`
+	getProductImageQuery = `SELECT url from productimages where product_id=$1`
+)
+
+func (s *pgStore) GetCart(ctx context.Context, user_id int) (cart_products []CartProduct, err error) {
+	var pids, quantities []int
+	var products []Product
+
+	err = s.db.Select(&pids, getCartQuery, user_id)
```

Comment:
1) handle err != nil , log error & return
2) Why are you using two query to get data from two tables? add a single query to fetch data from **cart**